test/libsolidity/SemanticTest.cpp

```diff
@@ -98,11 +98,18 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 			}
 			else
 			{
-				bytes output = callContractFunctionWithValueNoEncoding(
-					test.call().signature,
-					test.call().value,
-					test.call().arguments.rawBytes()
-				);
+				bytes output;
+				if (!test.call().hasCalldata)
+					if (test.call().hasValueSet)
```

Comment:
Do you actually need this flag? Can't you just use ``callFallbackWithValue`` with ``0`` as argument (which should be the default value of ``test.call().value`` if no value was set)?

Reply:
If that still works for selecting either `fallback {}` or `ether {}`, I'm fine with removing this flag.

Reply:
IMO you don't want to select which function it is in the test expectations anyways, but just want to specify msg.data and msg.value directly - the test expectations shouldn't be smart about such stuff, otherwise we'll need have more tests for them :-). In any case, it will be enough anyways.